src/ray/gcs/subscribe_executor.h

```diff
@@ -0,0 +1,91 @@
+#ifndef RAY_GCS_SUBSCRIBE_EXECUTOR_H
+#define RAY_GCS_SUBSCRIBE_EXECUTOR_H
+
+#include <atomic>
+#include <mutex>
+#include "ray/gcs/callback.h"
+#include "ray/gcs/tables.h"
+
+namespace ray {
+
+namespace gcs {
+
+/// \class SubscribeExecutor
+/// SubscribeExecutor class encapsulates the implementation details of
+/// subscribe or unsubscribe to any operations of elements (actors or tasks or objects).
+/// Either subscribe to specific elements or subscribe to all.
+template <typename ID, typename Data, typename Table>
+class SubscribeExecutor {
```

Comment:
I'm trying to think of a better name here... would something like `SubscriptionManager` be better here?

Reply:
ok, can rename to `SubscriptionExecutor` then?

Reply:
haha, ok.